### PR TITLE
feat(PUPIL-285): add copyright notice to lesson `PupilIntro`

### PR DESCRIPTION
**Pull request by carlmw**
Music year: [1975](https://www.youtube.com/watch?v=dAoaVU3-ve0)

Displays the legacy copyright notice for legacy lessons and the current license for new lessons

## How to test

1. Go to https://deploy-preview-2243--oak-web-application.netlify.thenational.academy/pupils/programmes/maths-secondary-ks3/units/graphical-representations-of-data/lessons/constructing-bar-charts-by-utilising-technology
2. Click on "Intro"
3. You should see the correct licence and links

### Desktop
localhost_3000_pupils_programmes_maths-secondary-ks3_units_graphical-representations-of-data_lessons_constructing-bar-charts-by-utilising-technology (1)

### Tablet
localhost_3000_pupils_programmes_maths-secondary-ks3_units_graphical-representations-of-data_lessons_constructing-bar-charts-by-utilising-technology(iPad Air) (2)

### Mobile
localhost_3000_pupils_programmes_maths-secondary-ks3_units_graphical-representations-of-data_lessons_constructing-bar-charts-by-utilising-technology(iPhone 14 Pro Max) (2)

### Legacy licence copy
Screenshot 2024-02-13 at 10 41 23

### CODE REVIEW

File: src/components/PupilComponents/CopyrightNotice/CopyrightNotice.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    );
+  }
+
+  return (
+    <OakP $font="body-4" $color="text-subdued">
+      This content is © Oak National Academy Limited (2023), licensed on{" "}
+      <OakLink
+        href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+        rel="external"
+      >
```
Comment (carlmw) [inline src/components/PupilComponents/CopyrightNotice/CopyrightNotice.tsx:40]: The link for the OGL doesn't automatically open in a new tab. I can of course make it do that but personally I trust users to know how to do it themselves or to be able to use the back button. This is the hill I will die on.

File: src/components/PupilComponents/CopyrightNotice/CopyrightNotice.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+type CopyrightNoticeProps = {
+  isLegacyLicense: boolean;
+};
+
+/**
+ * Displays the copyright notice for a lesson.
+ * `isLegacyLicense` indicates whether the lesson is using the current or old license.
+ */
+export const CopyrightNotice = ({ isLegacyLicense }: CopyrightNoticeProps) => {
```
Comment (carlmw) [inline src/components/PupilComponents/CopyrightNotice/CopyrightNotice.tsx:14]: I pinched the copy for the legacy licence from the teachers section

File: src/components/PupilViews/PupilIntro/PupilIntro.view.tsx [hunk 3]
```diff
@@ -81,109 +85,118 @@ export const PupilViewsIntro = (props: PupilViewsIntroProps) => {
...
+        $maxWidth={["100%", "all-spacing-22", "100%"]}
+        $mb={["space-between-none", "space-between-s"]}
+        $mh="auto"
+        $ph={["inner-padding-m", "inner-padding-xl", "inner-padding-none"]}
+        $minHeight="100%"
+        $gridTemplateRows={[
+          "min-content min-content 1fr",
+          "min-content min-content 1fr",
+          "min-content 1fr min-content",
+        ]}
```
Comment (carlmw) [inline src/components/PupilViews/PupilIntro/PupilIntro.view.tsx:99]: This ensures that the licence sticks to the bottom of the page while allowing the page to scroll if needed

**Comment by messmike** [comment]
One minor thing that I noticed was that the links don't take you back to the intro page, but lands back on the overview page. If that's expected/has to be that way then feel free to ignore.

**Comment by carlmw** [comment]
> One thing that I noticed was that the links don't take you back to the intro page, but lands back on the overview page. If that's expected/has to be that way then feel free to ignore.

when you return with the back button? If so yes, that is a known limitation of the current approach. There is a placeholder ticket to improve things to make proper navigation (including the back button) work as you'd expect https://www.notion.so/oaknationalacademy/23-24-Winter-OKRs-SPRINT-8-bc6cb4b6cc884ef5b29e911fff0cfe09?p=e3a6231612e94283b89d0805f004ad5c&pm=s

**Review by messmike — APPROVED** [review]

**Review by kimon-satan — APPROVED** [review]

**Pull request merged**